pro/src/pages/IndividualOffer/IndividualOfferDetails/commons/utils.ts

```diff
@@ -97,32 +86,58 @@ export const completeSubcategoryConditionalFields = (
     ...(subcategory?.isEvent ? ['durationMinutes'] : []),
   ] as (keyof DetailsFormValues)[]
 
-export const formatVenuesOptions = (
-  venues: VenueListItemResponseModel[],
-  isOnline: boolean
-) => {
-  //  We want to display the virtual venues only if there are no physical venues available
-  //  We also want to prevent selecting a virtual venue for a physical offer form
+/**
+ * **Rules**
+ * 1. Virtual venues are shown **only when no physical venue exists**.
+ * 2. When this is a **physical** offer (`isOfferVirtual === false`), virtual venues are never shown.
+ */
+export function filterAvailableVenues(
+  venues: readonly VenueListItemResponseModel[],
+  isOfferVirtual: boolean
+): VenueListItemResponseModel[] {
   const hasAtLeastOnePhysicalVenue = venues.some((v) => !v.isVirtual)
+  // Virtual venues are allowed only if *all* venues are virtual *and* the offer is virtual.
+  const shouldIncludeVirtualVenues =
+    !hasAtLeastOnePhysicalVenue && isOfferVirtual
+
+  return venues.filter((v) => shouldIncludeVirtualVenues || !v.isVirtual)
+}
+
+export function getVenuesAsOptions(
+  venues: readonly VenueListItemResponseModel[]
+): Option[] {
   return venues
-    .filter((venue) =>
-      hasAtLeastOnePhysicalVenue || !isOnline ? !venue.isVirtual : true
-    )
-    .map((venue) => ({
-      value: venue.id.toString(),
-      label: computeVenueDisplayName(venue),
+    .map((v) => ({
+      value: String(v.id),
+      label: computeVenueDisplayName(v),
     }))
     .sort((a, b) => a.label.localeCompare(b.label, 'fr'))
 }
 
-type SetDefaultInitialValuesFromOfferProps = {
-  offer: GetIndividualOfferResponseModel
-  subcategories: SubcategoryResponseModel[]
+export function getInitialValuesFromVenues(
+  availableVenues: VenueListItemResponseModel[],
+  isNewOfferCreationFlowFeatureActive: boolean
+) {
+  //  When there is only one venue available, we can automatically use this venue as the initially selected one.
+  const onlyVenue =
+    availableVenues.length === 1 ? availableVenues[0] : undefined
+  const venueId = onlyVenue?.id.toString() ?? ''
+
+  const maybeAccessibility = isNewOfferCreationFlowFeatureActive
+    ? getAccessibilityInfoFromVenue(onlyVenue).accessibility
+    : undefined
+
+  return {
+    ...DEFAULT_DETAILS_FORM_VALUES,
+    venueId,
+    ...(maybeAccessibility ? { accessibility: maybeAccessibility } : {}),
+  }
```

Comment:
Est-ce que ça serait pas plus simple ainsi ? (ça évite deux ternaires)

```suggestion
  //  When there is only one venue available, we can automatically use this venue as the initially selected one.
  const onlyVenue =
    availableVenues.length === 1 ? availableVenues[0] : undefined
  const venueId = onlyVenue?.id.toString() ?? ''
  
  if (isNewOfferCreationFlowFeatureActive) {
     return {
        ..DEFAULT_DETAILS_FORM_VALUES,
        venueId,
        accessibility: getAccessibilityInfoFromVenue(onlyVenue).accessibility,
     }
  }

  return {
    ...DEFAULT_DETAILS_FORM_VALUES,
    venueId,
  }
```